src/InputFilter/RestService/PatchInputFilter.php

```diff
@@ -75,7 +75,7 @@ public function init()
             'continue_if_empty' => true,
             'validators'        => [
                 new CallbackValidator(function ($value) {
-                    if (is_numeric($value) && intval($value) !== $value) {
+                    if (! is_numeric($value) || ((string) $value) !== (string) (int) $value) {
```

Comment:
I think we need a comment here indicating _why_ the casting is being done the way it is. At first glance, it looks like a mistake.

I almost wonder if this should be:

```
if (is_numeric($value) && intval($value) != $value) {
```

with a potential CS and/or Psalm rule suppression preceding, as what we're trying to determine is if we got an integer-ish value.

Reply:
@weierophinney I added a comment here to make things clearer. I think @Ocramius got a point here on not wanting to rely on a non-strict comparison. Let me know what you decide :-)